### ISSUE TITLE
3.3.1 should revert back to previous retrolambda version because 2.3.1 causes bad code to be generated for some lambdas...

### ISSUE THREAD

**Issue by kenyee** [evant/gradle-retrolambda#219]
FYI...in case anyone else gets bit by this:
https://github.com/orfjackal/retrolambda/issues/115

**Comment by evant** [comment]
You can downgrade the version yourself with
```
dependencies {
  retrolambdaConfig 'net.orfjackal.retrolambda:retrolambda:2.2.0'
}
```
I'll think about changing the default, ideally this should be fixed upstream though.

**Comment by sp-1234** [comment]
>ideally this should be fixed upstream though.

from upstream:
>Latest commit 93ff6ad on **30 Apr**

and today is November…  
I doubt that we live in an ideal world, at least right now.

**Comment by kenyee** [comment]
The config param works fine to include an older version...just wanted to flag this in case anyone else hit this issue...

**Comment by AlexMatiash** [comment]
I've just hit this wall.
@evant Thank you for the solution.

**Comment by rpattabi** [comment]
FYI. https://github.com/orfjackal/retrolambda/issues/115 is fixed now. This issue may be closed.

**Comment by sp-1234** [comment]
shouldn't it be closed when a new version is released which uses retrolambda 2.5?

**Comment by kenyee** [comment]
Oops...that's a fair point @sp-1234 :-)

**Comment by evant** [comment]
Fixed in `3.5.0`
